js/ui/html_editor/modules/imageUpload.js

```diff
@@ -71,11 +71,7 @@ if(Quill) {
 
         _onUploaded(data) {
             const { index: pasteIndex } = this.quill.getSelection() ?? { index: this.quill.getLength() };
-
-            const imageUrl = correctSlashesInUrl(this.options.uploadDirectory) + data.file.name;
-            urlUpload(this.quill, pasteIndex, { src: imageUrl });
-
-            this.quill.setSelection(pasteIndex + 1, 0);
+            serverUpload(this.options.uploadDirectory, data.file.name, this.quill, pasteIndex);
```

Comment:
After your change we don't call `setSelection` with third parameter = "api"`anymore. Is it expected?

Reply:
`setSelection` called inside `serverUpload` method. But there is no `api` parameter. I don't see this parameter anywere...

Reply:
third parameter is `api` by default

Reply:
According to the https://quilljs.com/docs/api/#events description:

> "Changes may occur through an API but as long as they originate from the user, the provided source should still be "user""

Thus, when image is dropped by user there should be 'user' display in events, not 'api'. And new changes more correct in this case I think